llvm/lib/CodeGen/LiveDebugValues/InstrRefBasedImpl.h

```diff
@@ -1200,21 +1200,21 @@ class InstrRefBasedLDV : public LDVImpl {
   /// exists, otherwise returns std::nullopt.
   std::optional<ValueIDNum> getValueForInstrRef(unsigned InstNo, unsigned OpNo,
                                                 MachineInstr &MI,
-                                                const ValueTable *MLiveOuts,
-                                                const ValueTable *MLiveIns);
+                                                const FuncValueTable *MLiveOuts,
+                                                const FuncValueTable *MLiveIns);
```

Comment:
Given the old patch passed a pointer to the start of a contiguous sequence, it made sense for it to be a pointer. But the new thing is passing a pointer to a sequence, not the start of it, so I think maybe /this/ patch is the one where this should change from a pointer to a reference, rather than waiting to do that in a follow-on patch.

Reply:
Ah, OK - optionality makes sense, thanks!